code-challenges.test.js

```diff
@@ -15,24 +17,59 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+describe('arrayShuffled', () => {
+    it('takes in an array, removes the first item from the array and shuffles the remaining items.', () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+    expect(arrayShuffled(colors1)).toContain("blue", "green", "yellow", "pink")
+    expect(arrayShuffled(colors2)).toContain("indigo", "periwinkle", "ochre", "aquamarine", "saffron")
+    })
+})
+
+// ReferenceError: arrayShuffle is not defined
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// input: Array of colors
+// output: Array of colors with the first item removed from the array and shuffles the remaining
+// create a function that takes in all the variables with arrays
+const arrayShuffled = (colors) => { 
+    // create a new variable with the given variable without the first element in the array.
+    let shuffled = colors.splice(1) 
+    // assigns a random 'sort' value to each color
+    .map(colors => ({ colors, sort: Math.random()}))
+    // sorts the colors array with a new mapped index based on the new values that came from the previous line
+    .sort((a, b) => a.sort - b.sort)
+    .map(({ colors }) => colors)
+    // returns the shuffled array of colors
+    return shuffled
```

Comment:
This is a great solution, AC!  I would love to see an explanation of what you're doing in lines 43 - 46 so I know you understand.  Great pseudocode! Since we didn't cover this approach in class, I'm concerned you might have found this solution elsewhere, which is fine- just let us know that you did and please explain the process after the function next time.

Reply:
Hi Trish! I definitely pulled that code off a website and just applied what I could/needed. I will make sure to insert additional information on my next assessments if and when I do venture outside of the syllabus.  Thank you!